sql/core/src/main/scala/org/apache/spark/sql/execution/SparkQl.scala

```diff
@@ -42,6 +45,95 @@ private[sql] class SparkQl(conf: ParserConf = SimpleParserConf()) extends Cataly
           getClauses(Seq("TOK_QUERY", "FORMATTED", "EXTENDED"), explainArgs)
         ExplainCommand(nodeToPlan(query), extended = extended.isDefined)
 
+      case Token("TOK_REFRESHTABLE", nameParts :: Nil) =>
+        val tableIdent = extractTableIdent(nameParts)
+        RefreshTable(tableIdent)
+
+      case Token("TOK_CREATETABLEUSING", createTableArgs) =>
+        val Seq(
+          temp,
+          allowExisting,
+          Some(tabName),
+          tableCols,
+          Some(Token("TOK_TABLEPROVIDER", providerNameParts)),
+          tableOpts,
+          tableAs) = getClauses(Seq(
+          "TEMPORARY",
+          "TOK_IFNOTEXISTS",
+          "TOK_TABNAME", "TOK_TABCOLLIST",
+          "TOK_TABLEPROVIDER",
+          "TOK_TABLEOPTIONS",
+          "TOK_QUERY"), createTableArgs)
+
+        val tableIdent: TableIdentifier = tabName match {
+          case Token("TOK_TABNAME", Token(dbName, _) :: Token(tableName, _) :: Nil) =>
```

Comment:
I am personally not big on using wildcards in pattern matches. This prevents us from catching a grammar problem early. Since most of the wildcards are actually empty lists, why not match these?